fedimint-core/src/db/mod.rs

```diff
@@ -2431,13 +2432,14 @@ mod tests {
         let val = TestVal(2);
         let db = Database::new(MemDatabase::new(), ModuleDecoderRegistry::default());
 
+        let key_task = waiter(&db.new_isolated(module_instance_id), TestKey(1)).await;
```

Comment:
Was this failing because the db wasn't isolated before?

Reply:
no, this doesn't fix the test. this makes the test actually fail.
because notification in moduledatabasetransaction don't work at all.